### ISSUE TITLE
FileNotFoundError when conducted the evaluation

### ISSUE THREAD

**Issue by SiyuanHuang95** [autonomousvision/plant#7]
Hi, thanks for sharing, and quite interesting work!

I tried your code base and the whole training process is without errors. However, when I finished training and used the commands to do the evaluations,

```
python leaderboard/scripts/run_evaluation.py user=$USER experiments=PlanTmedium3x eval=longest6
```
error occurs with:

```
 [Errno 2] No such file or directory: '/XXX/plant/checkpoints/PlanT/3x/PlanT_medium/log/args.txt'
```
I check the source code, find we need to configure the config path to find args.txt,  from the line [Here](https://github.com/autonomousvision/plant/blob/0559385f06b3709e4c11d4ff7c9ef6d24ce71e35/carla_agent_files/PlanT_agent.py#L39).

Did I miss something important?

**Comment by RenzKa** [comment]
Hei,
Did you change the path to your model in the config file?(Here: https://github.com/autonomousvision/plant/blob/0559385f06b3709e4c11d4ff7c9ef6d24ce71e35/carla_agent_files/config/experiments/PlanTmedium3x.yaml#L9)

**Comment by SiyuanHuang95** [comment]
Nopes. 

It seems that it should be pointed to the 

```
training/DEBUG/model=PlanT_user=XXX/xxxx.
```

when using the default setting for training

**Comment by RenzKa** [comment]
Hi,
we made some initial tests for choosing the epoch and kept it fixed afterwards. However, we found that a lower epoch number (e.g. 11) results in similar results (just slightly worse) but more robust (i.e., lower variance) over many seeds.

The evaluation process support only one Carla server.  If multiple GPUs are available you can split the evaluation over routes (i.e., call the evaluation script independently for each route or a subset of routes and spawn one server for each execution - you also need to change the port and the traffic-manager port). We provide a split version of the evaluation routes [here](https://github.com/autonomousvision/plant/tree/main/leaderboard/data/longest6/longest6_split).

**Comment by SiyuanHuang95** [comment]
Hi,

Thanks for your update. The ckpt selection is quite interesting and very useful;  thanks! Have you also conducted some other epoch selection experiments that we can share with us? Also, I have noticed that evaluating the Imitation learning-based methods is quite tricky; some papers chose to evaluate many epochs and report the best one. Do you think the shifting problem in IL is still big? BTW, did the seed affect the Carla result also, or did the seed here only affect the model infer?

Thanks for the insights on the evaluation part.


Bests,

**Comment by RenzKa** [comment]
Hi,

1. For the final results, I did not evaluate many epochs. I fixed the epoch and always reported the result for this epoch. Evaluating many epochs requires much more compute and might lead to 'overfitting' when done on the test distribution. Since the diversity in Carla is currently quite limited (number of towns etc.) this could lead indeed to a problem if we would evaluate on a different distribution.

2.  "did the seed affect the Carla result also, or did the seed here only affect the model infer?" 
I am not 100% sure if I understood your question correctly.
  The Carla results are dependent on the model inference. So if the model inference changes, the CARLA results are also different. What exactly do you mean by 'Carla results' and by 'model infer'?

**Comment by SiyuanHuang95** [comment]
All clear. 

Thanks for your information and patient reply. 


Bests,
